src/main/java/org/dependencytrack/persistence/ComponentQueryManager.java

```diff
@@ -136,33 +136,39 @@ public List<Component> getAllComponents(Project project) {
      * @return a List of Dependency objects
      */
     public PaginatedResult getComponents(final Project project, final boolean includeMetrics) {
-        return getComponents(project, includeMetrics, false);
+        return getComponents(project, includeMetrics, false, false);
     }
     /**
      * Returns a List of Dependency for the specified Project.
      * @param project the Project to retrieve dependencies of
      * @param includeMetrics Optionally includes third-party metadata about the component from external repositories
      * @param onlyOutdated Optionally exclude recent components so only outdated components are shown
+     * @param onlyDirect Optionally exclude transitive dependencies so only direct dependencies are shown
      * @return a List of Dependency objects
      */
-    public PaginatedResult getComponents(final Project project, final boolean includeMetrics, final boolean onlyOutdated) {
+    public PaginatedResult getComponents(final Project project, final boolean includeMetrics, final boolean onlyOutdated, final boolean onlyDirect) {
         final PaginatedResult result;
         String querySring ="SELECT FROM org.dependencytrack.model.Component WHERE project == :project ";
         if (filter != null) {
             querySring += " && (project == :project) && name.toLowerCase().matches(:name)";
         }
         if (onlyOutdated) {
+            // Components are considered outdated when metadata does exists, but the version is different than latestVersion
+            // Different should always mean version < latestVersion
```

Comment:
@walterdeboer @nscuro have we given any thought to latest version in a repo may be a BETA vs "prod/GA/GTM/stable" code?   if the version we have is the last stable release, but the repo is showing a beta release...do we want to flag that as outdated?

Reply:
@melba-lopez That's a good point, and I fear it is somewhat subjective. That said, the focus is on flagging risk, so the question becomes "am I at risk of being outdated when there's a new BETA version available?". Unless the latest stable version has risky vulns (which would be flagged separately), relying on BETA software in itself would be an operational risk. 

What @walterdeboer is trying to solve here is risk being flagged despite there not being any sane way forward (no stable version to update to).

An extension of this feature could be to capture both latest stable and latest unstable version from remote repositories, and then letting the users decide what they want to consider "latest".